packages/wdio-browserstack-service/src/service.ts

```diff
@@ -86,29 +93,34 @@ export default class BrowserstackService implements Services.ServiceInstance {
         this._scenariosThatRan = []
 
         if (this._observability && this._browser) {
-            this._insightsHandler = new InsightsHandler(
-                this._browser,
-                this._isAppAutomate(),
-                this._config.framework
-            )
-            await this._insightsHandler.before()
-
-            /**
-             * register command event
-             */
-            this._browser.on('command', async (command) => await this._insightsHandler?.browserCommand(
-                'client:beforeCommand',
-                Object.assign(command, { sessionId: this._browser?.sessionId }),
-                this._currentTest
-            ))
-            /**
-             * register result event
-             */
-            this._browser.on('result', async (result) => await this._insightsHandler?.browserCommand(
-                'client:afterCommand',
-                Object.assign(result, { sessionId: this._browser?.sessionId }),
-                this._currentTest
-            ))
+            try {
+                this._insightsHandler = new InsightsHandler(
+                    this._browser,
+                    this._isAppAutomate(),
+                    this._config.framework
+                )
+                await this._insightsHandler.before()
```

Comment:
The constructor of insight handler can also raise an error.

The whole block of code was wrapped in try catch to ensure nothing breaks the tests from running even if there is unforeseen error in observability code. So even if, there is no chance of breaking now, but future code addition shouldn't break the user tests. So would like to wrap the entire code in try catch just to ensure that it doesn't break any user flow.

Reply:
Nothing that breaks in the `before` hook will cause the test to break. You will have to throw a `SevereServiceError` to make the hook actual stop the test. The event listener calls can be moved outside of the `try/catch` because it is impossible that they throw.